src/geo/gmaps/gmaps-map-view.js

```diff
@@ -67,6 +67,7 @@ var GoogleMapsMapView = MapView.extend({
     });
 
     google.maps.event.addListener(this._gmapsMap, 'bounds_changed', function (e) {
+      self.tileErrorCollection.resetErrorTiles();
```

Comment:
What about not adding this logic within gmaps or leaflet code? Check if it is feasible using it map attributes (center, bounds,...) change.